docs: Add a note about package requirements for fact gathering (#70796)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_vars_facts.rst

You can use facts in conditionals (see :ref:`playbooks_conditionals`) and also in templates. You can also use facts to create dynamic groups of hosts that match particular criteria, see the :ref:`group_by module <group_by_module>` documentation for details.

.. _fact_requirements:

Package requirements for fact gathering
---------------------------------------

> For fact gathering, certain binaries are used. If binary is absent, fact is set to default value.

Reviewer: ```suggestion
Some fact gathering depends on installed binaries on the remote host. If the binary is absent, the fact is set to a default value.
```

> It is recommended to install these binaries on the remote host using default OS pacakge manager.

Reviewer: ```suggestion
We recommend that you install these binaries on the remote host using default OS package manager.
```

> Following is a list of such requirements:

Reviewer: ```suggestion
The following lists required binaries:
```

> * For Linux Network fact gathering, ``ip`` binary is used which comes from ``iproute2`` package.

Reviewer: ```suggestion
* Linux Network fact gathering -  Depends on  the ``ip`` binary from the ``iproute2`` package.
```

> * Linux Network fact gathering -  Depends on  the ``ip`` binary from the ``iproute2`` package.

Reviewer 2: this might not be correct name for all distros
